CMakeLists.txt

```diff
@@ -125,6 +126,10 @@ else()
     elseif(${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
         set(PLATFORM_LINUX TRUE CACHE INTERNAL "Target platform: Linux")
         message("Target platform: Linux " ${ARCH})
+    elseif(${CMAKE_SYSTEM_NAME} STREQUAL "FreeBSD")
+        set(PLATFORM_LINUX TRUE CACHE INTERNAL "Target platform: Linux")
+        set(PLATFORM_FREEBSD TRUE CACHE INTERNAL "Target platform: FreeBSD")
```

Comment:
Defining both `PLATFORM_LINUX` and `PLATFORM_FREEBSD` is a bit inconsistent with the rest of platform variables as only a single variable is expected to be defined.
Will it build if you try to build at as Linux?

Reply:
This was simpler for me to do as the vast majority of the Linux stuffs in the build system translates pretty clearly to FreeBSD.  I don't mind rejiggering everything so only PLATFORM_FREEBSD is used, if you want me to do that.

Reply:
> I don't mind rejiggering everything so only PLATFORM_FREEBSD is used, if you want me to do that.

This will require quite a lot of changes, which are not really needed.
Heaving `PLATFORM_FREEBSD` and `PLATFORM_LINUX` at the same time may be OK.

Reply:
It should be fine, as PLATFORM_FREEBSD is only added to the environment if it's running on FreeBSD.  Functionally, all the Linux stuff is untouched.